### ISSUE TITLE
Question: DOM parser

### ISSUE THREAD

**Issue by maoamid** [eclipsesource/tabris-js#1163]
There is an equivalent for new DOMParser(); in tabris? 
I need to parse some html string, and regex isn't cover my all situations.

**Comment by cookieguru** [comment]
I can confirm that [https://www.npmjs.com/package/cheerio](Cheerio) works great.

**Comment by maoamid** [comment]
@cookieguru thank you a lot, I'll try it

**Comment by maoamid** [comment]
@cookieguru are You using Cheerio with beta version? For me, it works, but it's slow down my app when it is called first time, app display a message that is not responding, and after a while (about 20 sec) it does what is supposed to do, if I tap wait. I didn't  test a lot

**Comment by cookieguru** [comment]
@maoamid Yes, I'm noticing that behavior now too, though I hadn't noticed it before today.  I'm not running the latest and greatest versions, either:
```plain
myapp
├─┬ cheerio@0.22.0
│ ├─┬ css-select@1.2.0
│ │ ├── boolbase@1.0.0
│ │ ├── css-what@2.1.0
│ │ ├── domutils@1.5.1
│ │ └── nth-check@1.0.1
│ ├─┬ dom-serializer@0.1.0
│ │ └── domelementtype@1.1.3
│ ├── entities@1.1.1
│ ├─┬ htmlparser2@3.9.2
│ │ ├── domelementtype@1.3.0
│ │ ├── domhandler@2.3.0
│ │ ├── inherits@2.0.1
│ │ └─┬ readable-stream@2.2.2
│ │   ├── buffer-shims@1.0.0
│ │   ├── core-util-is@1.0.2
│ │   ├── isarray@1.0.0
│ │   ├── process-nextick-args@1.0.7
│ │   ├── string_decoder@0.10.31
│ │   └── util-deprecate@1.0.2
│ ├── lodash.assignin@4.2.0
│ ├── lodash.bind@4.2.1
│ ├── lodash.defaults@4.2.0
│ ├── lodash.filter@4.6.0
│ ├── lodash.flatten@4.4.0
│ ├── lodash.foreach@4.5.0
│ ├── lodash.map@4.6.0
│ ├── lodash.merge@4.6.0
│ ├── lodash.pick@4.4.0
│ ├── lodash.reduce@4.6.0
│ ├── lodash.reject@4.6.0
│ └── lodash.some@4.6.0
├── moment@2.17.1
├── moment-timezone@0.5.11
├── tabris@1.10.0
└─┬ tabris-js-node@0.1.0
  ├─┬ assert@1.4.1
  │ └── util@0.10.3
  ├─┬ buffer@3.6.0
  │ ├── base64-js@0.0.8
  │ ├── ieee754@1.1.6
  │ └── isarray@1.0.0
  ├─┬ console-browserify@1.1.0
  │ └── date-now@0.1.4
  ├── constants-browserify@0.0.1
  ├── process@0.10.1
  └─┬ timers-browserify@1.4.2
    └── process@0.11.5
```
It was working fine [when I first started using it](http://www.github.com/eclipsesource/tabris-js/issues/921) so I'm not sure what happened.

**Comment by cookieguru** [comment]
@maoamid I just popped a few lines in to time the execution and here's what I noticed:

```js
//start timer
require('tabris-js-node'); //2 seconds elapsed since start
var cheerio = require('cheerio'); //18.337 seconds elapsed since start
var $ = cheerio.load(htmlString); // 18.457 seconds elapsed since start
//perform a bunch of traversals and edit attributes of various elements // 18.512 seconds elapsed since start
$.html(); // 18.525 seconds elapsed since start
```
The first time I ran the code above it clocked in at 25 seconds.  Not sure why the second run was significantly faster.

Obviously loading Cheerio takes the most amount of time, so I ran `tail -f` on my HTTP access log while the UI in the app was blocked.  It scrolled by like level 73 of Tetris :)

I ran the same code in Node and the entire block took 0.25 seconds.  So I'd be willing to bet that the problem is just that this is a bloated library that takes forever to transfer all the files over HTTP.  When you actually build the app, it'll just be looking at the local filesystem and _should_ be *much* faster.  I have not yet built this app so I can't say for certain.

If your requirements are just pulling things out of a webpage, you might look at one of Cheerio's dependencies: `htmlparser2`.  However my use case was modifying the contents prior to displaying them inside a `WebView` and Cheerio was the only library I could find that fit my needs.
